Gutenboarding: Accordion-style plans grid.

yansern opened the pull request:
## Changes proposed in this Pull Request

* Accordion-style plans grid.

**Detailed Changes:**

- We now split the rendering of PlanItem, one for primary plan, one for all other plans (it used to be in a single loop). Primary plan could either mean the default premium plan or the recommended plan based on the features user chosen from the feature picker. See https://github.com/Automattic/wp-calypso/pull/44299/files#diff-ef06ad3b1ea3e3d50b5d84d92780d2adR39-R48.
  - The `badge` param is introduced to customize what to show on the primary plan's badge. When default premium plan is shown, the badge display **Popular**. When a recommended plan is shown, the badge displays **Recommended for you**.
- All plans toggling (Show all plans/collapse all plans button) is now the responsibility of the parent component `PlansTable` instead of child component `PlanItem`, see https://github.com/Automattic/wp-calypso/pull/44299/files#diff-ef06ad3b1ea3e3d50b5d84d92780d2adR50-R63. `onToggleExpandAll` and `allPlansExpanded` param has been removed, see https://github.com/Automattic/wp-calypso/pull/44299/files#diff-ef06ad3b1ea3e3d50b5d84d92780d2adR50-R63.
- "Custom domain not included" is now always shown under Free plan's feature list.
- Added plan descriptions "Best for freelancers, best for online stores, etc.".
- PlanItem component at `plan-tables\plan-item.tsx` is now moved out to `plan-item\index.tsx` with it's own stylesheet specific to the component itself.


## Testing instructions

* Go to `/new`.
* Test plans grid in different viewport sizes, make sure they match similarly to design.
* Test toggling "Show all plans / Collapse all plans" button.
  * Manually expanding/collapsing each of the plan items should cause the toggle button to automatically switch "Show all plans" to "Collapse all plans" (and vice versa).
* Test the domain picker button on the feature list with:
  * No/free domain selected.
  * Paid domain selected.
* Ensure all "Choose" verb has been renamed to "Select".

## Screenshots
![2020-07-22_17-07-41](https://user-images.githubusercontent.com/1287077/88193543-16284d00-cc3e-11ea-934d-0400cb7a476a.png)
![2020-07-22_17-08-03](https://user-images.githubusercontent.com/1287077/88193547-17f21080-cc3e-11ea-9667-2186a90a8ba8.png)
![2020-07-22_17-08-51](https://user-images.githubusercontent.com/1287077/88193565-1a546a80-cc3e-11ea-88db-637da34683dd.png)
![2020-07-22_17-09-06](https://user-images.githubusercontent.com/1287077/88193568-1b859780-cc3e-11ea-836c-53a743e0cd45.png)


Fixes #44201

a8ci18n commented on client/landing/gutenboarding/onboarding-block/plans/index.tsx:
```diff
@@ -73,10 +73,10 @@ const PlansStep: React.FunctionComponent< Props > = ( { isModal } ) => {
 				<SubTitle>
 					{ __(
-						'Pick a plan that’s right for you. Switch plans as your needs change. There’s no risk, you can cancel for a full refund within 30 days.'
+						'Pick a plan that’s right for you. There’s no risk, you can cancel for a full refund within 30 days.'
```
:warning: This change will be queued for retranslation. We'll display the English text until then.

yansern commented:
**Things left to work on:**

1. Full width badge on mobile view.
![image](https://user-images.githubusercontent.com/1287077/88030728-e6911c00-cb3b-11ea-89fa-5f4e82876fa1.png)

2. Intended audience caption.
![image](https://user-images.githubusercontent.com/1287077/88030852-0a546200-cb3c-11ea-88a9-80c753018882.png)

michelleweber commented:
Re: copy, the captions should be in sentence case, with a capital letter for the first word only.

Have we ever tested using "recommended" vs. "best" for these?

razvanpapadopol commented:
Few more things related to design:
1. Increase height of the domain checkmark/button to align with the rest of the items.
<img width="497" alt="Screenshot 2020-07-21 at 16 10 34" src="https://user-images.githubusercontent.com/14192054/88059022-f117d980-cb6c-11ea-8b57-4229689c6703.png">

2. Apply primary button styles to the buttons at the bottom of _Detailed comparison_ table. They should look similar to the buttons used on plans cards (rounded corners, padding, etc).
<img width="601" alt="Screenshot 2020-07-21 at 15 56 58" src="https://user-images.githubusercontent.com/14192054/88059302-566bca80-cb6d-11ea-99c0-6d962a618f81.png">

yansern commented:
> Re: copy, the captions should be in sentence case, with a capital letter for the first word only.

Plan description (intended audience caption) is now in sentence case. https://github.com/Automattic/wp-calypso/pull/44299/commits/af57d4d7551261a51d1927acc817d45bc1f1fa5e

> Have we ever tested using "recommended" vs. "best" for these?

On the sidenote, from the figma design I can see there are some additional mechanics around this which is tied to the feature picker.

When a user picks a set of features, the badge for the primary plan will show "Recommended For You".
When a user doesn't pick anything, the badge for the recommended plan will show "Popular".

See: https://github.com/Automattic/wp-calypso/pull/44299/files#diff-ef06ad3b1ea3e3d50b5d84d92780d2adR40-R45

yansern commented:
> Increase height of the domain checkmark/button to align with the rest of the items.

Fixed with `min-height`. https://github.com/Automattic/wp-calypso/pull/44299/files#diff-3cc6363676acacebd55683fdfe4bfa99R357

> Apply primary button styles to the buttons at the bottom of Detailed comparison table. They should look similar to the buttons used on plans cards (rounded corners, padding, etc).

Fixed. https://github.com/Automattic/wp-calypso/pull/44299/commits/f65b19acf0ff6dcfb1efff7c4bbf1e001825b696

yansern commented:
I reworked the plans grid again based on closer inspection of the figma design, PR descriptions have been updated & screenshots have been replaced.

razvanpapadopol commented on packages/plans-grid/src/plan-item/index.tsx:
```diff
@@ -91,115 +103,120 @@ function domainMessageStateMachine(
-						}
-						className="plan-item__summary"
+						onClick={ handleToggle }
+						onKeyDown={ ( e ) => e.keyCode === SPACE_BAR_KEYCODE && handleToggle() }
```
Why not use also _ENTER_ key to toggle accordion as mentioned in [WAI-ARIA spec](https://www.w3.org/TR/wai-aria-practices/)?

yansern replied:
And also with focus outline (not sure if there is at the moment). Another PR to fix A11y issues.

razvanpapadopol commented on packages/plans-grid/src/plans-table/index.tsx:
```diff
@@ -29,32 +31,85 @@ const PlansTable: React.FunctionComponent< Props > = ( {
+
+	// Primary plan
+	// TODO: Feature picker determines recommended plan depending on the features that user choose.
+	const recommendedPlanSlug = null;
```
This should be passed as a prop to plans-grid package and then passed to this component.

yansern replied:
Coming soon!

razvanpapadopol commented on packages/plans-grid/src/plans-table/index.tsx:
```diff
@@ -29,32 +31,85 @@ const PlansTable: React.FunctionComponent< Props > = ( {
+	// Primary plan
+	// TODO: Feature picker determines recommended plan depending on the features that user choose.
+	const recommendedPlanSlug = null;
+	const popularPlanSlug = 'value_bundle';
```
We should get this value from the plans data-store.
`useSelect( ( select ) => select( PLANS_STORE ).getDefaultPaidPlan() );`

yansern replied:
Going to roll this together with another PR coming up where `useRecommendedPlan()` is used.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4128953

Thank you @yansern for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4128953).
